config/Dockerfiles/tests.d/libvirt/02_libvirt-storage

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash -xe
+
+# Verify basic provisioning for all supplied providers
+# distros.exclude: fedora26 centos7 fedora28
+# providers.include: libvirt
+# providers.exclude: none
+
+## NOTE: This is a temporary test to run libvirt tests until we fix
+## the networking race condition. Tests will only run on centos7.
+
+DISTRO=${1}
+PROVIDER=${2}
+
+TEST_NAME="${DISTRO}/${PROVIDER}/02_libvirt-storage"
+DESCRIPTION="Test ${PROVIDER} provider with storage provisioning"
+
+echo "Test Name: ${TEST_NAME}"
+echo "${DESCRIPTION}"
+
+TARGET="libvirt-storage"
+
+pushd docs/source/examples/workspace
+
+function clean_up {
+    set +e
+    #linchpin -w . -p PinFile.${PROVIDER}.yml -v destroy ${TARGET}
+    # remove output file used to gen uhash if it exists
+    rm storage.txt || true
+}
+trap clean_up EXIT SIGHUP SIGINT SIGTERM
+
+linchpin -w . -p PinFile.${PROVIDER}.yml -v up ${TARGET} 2> storage.txt
```

Comment:
Consider performing a tee here. eg 

`linchpin -w . -p PinFile.${PROVIDER}.yml -v up ${TARGET} 2>&1 | tee -a storage.txt`

This will output to the display, and write to the log. Your current `2>` redirects STDERR.

Reply:
Be careful with pipes and return codes.  If you do add this tee then you better add the following to the top of the file
set -o pipefail
otherwise your return code will get swallowed up and this will only get the return code from tee which will be 0 unless it couldn't write to storage.txt